### PR TITLE
GCORE: Add support for GeoDNS/Failover

### PR DESCRIPTION
**Pull request by xddxdd**
Add metadata fields for configuring Gcore's GeoDNS and Healthcheck/Failover functionalities, as well as documentations for them.

Tested manually with my own config.

As a side note, I didn't add any automated tests because I'm not sure how we should do so for custom metadata fields. Should I add them as testgroups in `integrationTest/integration_test.go`?

### CODE REVIEW

**Comment by tlimoncelli** [comment]
Neat stuff!  Thanks for the PR!

Yes, please add integration tests.  Just limit them to GCORE similar to how Clouldflare's proxy tests are just for CLOUDFLAREAPI.

See this example around line 1964:

```
testgroup("CF_PROXY A create",
      only("CLOUDFLAREAPI"),
```

**Comment by xddxdd** [comment]
Added integration test, and made some small changes to the code to make it play nice with integ test.

File: providers/gcore/convertMetadata.go [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+	result := map[string]string{}
+
+	if len(rrset.Filters) > 0 {
+		result[metaFilters] = serializeRecordFilter(rrset.Filters)
+	}
+
+	// RRSet only supports failover field
+	if rrset.Meta != nil && rrset.Meta.Failover != nil {
+		failover := *rrset.Meta.Failover
+		// FIXME: maybe we can reduce duplication by using reflection here
```
Comment (tlimoncelli) [inline providers/gcore/convertMetadata.go:52]: > 		// FIXME: maybe we can reduce duplication by using reflection here

It's fine like it is.  A little repetition is better than a lot of complexity.

**Review by tlimoncelli — APPROVED** [review]

**Comment by tlimoncelli** [comment]
Great new feature!  I love the `withMeta()` function that makes testing so easy.

I've rebased.  Anything else before merging?

**Comment by xddxdd** [comment]
Looks good to me.

**Pull request merged**